### ISSUE TITLE
--gen step ignoring $ELASTICSEARCH_NODES

### ISSUE THREAD

**Issue by pieterlukasse** [opentargets/platform#494]
Contrary to the other steps in the pipeline, the --gen step seems to be ignoring the environment variable ELASTICSEARCH_NODES. Even though it is set to something else, the script still tries to contact the elasticsearch server at `localhost:9200`. 

See logs:
```
ConnectionError: ConnectionError(HTTPConnectionPool(host='localhost', port=9200): Max retries exceeded with url: /_bulk (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f3df80bd290>: Failed to establish a new connection: [Errno 111] Connection refused',))) caused by: ConnectionError(HTTPConnectionPool(host='localhost', port=9200): Max retries exceeded with url: /_bulk (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f3df80bd290>: Failed to establish a new connection: [Errno 111] Connection refused',)))
2019-03-19 12:14:42,987 mrtarget.common.ElasticsearchLoader ERROR - push to elasticsearch failed for chunk, giving up...
Traceback (most recent call last):
  File "mrtarget/common/ElasticsearchLoader.py", line 115, in flush
    self._flush()
  File "mrtarget/common/ElasticsearchLoader.py", line 140, in _flush
    stats_only=True)
  File "/usr/local/lib/python2.7/site-packages/elasticsearch/helpers/__init__.py", line 257, in bulk
    for ok, item in streaming_bulk(client, actions, **kwargs):
  File "/usr/local/lib/python2.7/site-packages/elasticsearch/helpers/__init__.py", line 192, in streaming_bulk
    raise_on_error, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/elasticsearch/helpers/__init__.py", line 99, in _process_bulk_chunk
    raise e
ConnectionError: ConnectionError(HTTPConnectionPool(host='localhost', port=9200): Max retries exceeded with url: /_bulk (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f3dfab7b5d0>: Failed to establish a new connection: [Errno 111] Connection refused',))) caused by: ConnectionError(HTTPConnectionPool(host='localhost', port=9200): Max retries exceeded with url: /_bulk (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f3dfab7b5d0>: Failed to establish a new connection: [Errno 111] Connection refused',)))
2019-03-19 12:14:45,852 mrtarget.modules.GeneData INFO - join workers
2019-03-19 12:15:01,199 mrtarget.common.Redis INFO - GeneObjectStorer-2 done processing
2019-03-19 12:15:02,280 mrtarget.common.Redis INFO - GeneObjectStorer-4 done processing
2019-03-19 12:15:02,280 mrtarget.common.Redis INFO - GeneObjectStorer-3 done processing
2019-03-19 12:15:02,280 mrtarget.common.Redis INFO - GeneObjectStorer-5 done processing
2019-03-19 12:15:04,444 mrtarget.modules.GeneData INFO - all gene objects pushed to elasticsearch
2019-03-19 12:15:36,857 __main__.main() INFO - `/usr/src/app/mrtarget/CommandLine.py mrtarget --release-tag 19.02 --gen --data-config /ot_files/opentargets-data-loading/load-from-json/docker-commands/mrtarget.data-2019-02-05....yml --gen-plugin-places mrtarget/plugins/gene` - finished
```

**Comment by pieterlukasse** [comment]
notify @afaulconbridge

**Comment by pieterlukasse** [comment]
discussion results: try to be explicit about the versions of the dependencies used instead of using latest version. E.g. in this case we suspect argparse library to have a bug. Such libraries should be included with a reference to a specific version.

**Comment by afaulconbridge** [comment]
There is now a pull request to pin dependencies, including transitive ones opentargets/data_pipeline#471 This should be included in 19.04 and then hopefully this issue will be reproducible

**Comment by pieterlukasse** [comment]
Update: I found the same error in --val step, but this had to do with a wrong --net parameter in my docker command which caused the code to fall back on localhost after failing to connect. Will check further, but it looks like a combination of issues in my script + the underlying elasticsearch library.
